File: api/src/test/java/org/pmiops/workbench/CallsRealMethodsWithDelay.java
package org.pmiops.workbench;

import org.mockito.internal.stubbing.answers.CallsRealMethods;
import org.mockito.invocation.InvocationOnMock;

public class CallsRealMethodsWithDelay extends CallsRealMethods {

  private final long delay;

  public CallsRealMethodsWithDelay(long delay) {
    this.delay = delay;
  }

  public Object answer(InvocationOnMock invocation) throws Throwable {
    Thread.sleep(delay);
    return super.answer(invocation);
  }

}

File: api/src/test/java/org/pmiops/workbench/CallsRealMethodsWithDelay.kt
package org.pmiops.workbench

import org.mockito.internal.stubbing.answers.CallsRealMethods
import org.mockito.invocation.InvocationOnMock

class CallsRealMethodsWithDelay(private val delay: Long) : CallsRealMethods() {

    @Throws(Throwable::class)
    override fun answer(invocation: InvocationOnMock): Any {
        Thread.sleep(delay)
        return super.answer(invocation)
    }
}
